src/DOM.js

```diff
@@ -175,6 +175,9 @@ angular.module('textAngular.DOM', ['textAngular.factories'])
 								}
 							}
 						}
+						_nodes = _nodes.filter(function(value, index, self) {
+							return self.indexOf(value) === index;
+						});
```

Comment:
So, you're only selecting the ones the it's index correspond to it's actual index? 
To prevent arrays populated like ```arr[10] = 1, arr[20] = 2```. Is that right?
If so, could you explain why that's necessary?